src/services/MarkdownRenderer.ts

```diff
@@ -129,7 +131,7 @@ export class MarkdownRenderer {
         this.currentTopHeading && this.currentTopHeading.id,
       );
     }
-    return this.originalHeadingRule(text, level, raw);
+    return this.originalHeadingRule(text, level, raw, this.slugger);
```

Comment:
Looks like you don't need to manually create `slugger`.
Just pass it through function parameters:

```ts
headingRule = (text: string, level: number, raw: string, slugger: marked.Slugger) => {
  // ...
  return this.originalHeadingRule(text, level, raw, slugger);
}
```

Reply:
Good catch, addressed.